Source/Controller/LaunchViewController.swift

```diff
@@ -82,105 +96,142 @@ class LaunchViewController: UIViewController {
-                        Analytics.shared.forget()
-                        CrashReporting.shared.forget()
-                        
-                        Task {
-                            await AppController.shared.relaunch()
-                        }
-                    }
+                if !isMigrating {
+                    // note that hmac key can be nil to switch it off
+                    guard let network = configuration.network else { return }
+                    guard let secret = configuration.secret else { return }
+                    guard let bot = configuration.bot else { return }
+                    
+                    try await bot.login(config: configuration)
                 }
-                controller.addAction(reset)
-
-                AppController.shared.showAlertController(with: controller, animated: true)
-                
-                return
+            } catch {
+                self.handleLoginFailure(with: error, configuration: configuration)
             }
             
-            self.launchIntoMain()
             
-            bot.about { (about, aboutErr) in
-                Log.optional(aboutErr)
-                // No need to show an alert to the user as we can fetch the current about later
-                CrashReporting.shared.reportIfNeeded(error: aboutErr)
-                
-                if let about = about {
-                    CrashReporting.shared.identify(
-                        identifier: about.identity,
-                        name: about.name,
-                        networkKey: network.string,
-                        networkName: network.name
-                    )
-                    Analytics.shared.identify(identifier: about.identity,
-                                              name: about.name,
-                                              network: network.name)
-                }
-            }
+            self.launchIntoMain()
+            await self.trackLogin(with: configuration)
         }
     }
+    
+    // MARK: - Helpers
 
-    private func launchIntoOnboarding(status: Onboarding.Status = .notStarted,
-                                      simulate: Bool = false) {
+    private func launchIntoOnboarding(
+        status: Onboarding.Status = .notStarted,
+        simulate: Bool = false
+    ) {
         DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
-            AppController.shared.showOnboardingViewController(status, simulate)
+            self.appController.showOnboardingViewController(status, simulate)
         }
     }
 
-    private func launchIntoMain() {
-
+    @MainActor private func launchIntoMain() {
         // no need to start a sync here, we can do it later
         // also, user is already logged in
 
         // transition to main app UI
-        DispatchQueue.main.async {
-            AppController.shared.showMainViewController(animated: true)
+        appController.showMainViewController(animated: true)
+    }
+
```

Comment:
The diff for this file got messy. The next two functions: `handleLoginFailure` and `trackLogin` don't have a lot of changes, I just factored them out of `launch`.